client/src/main/java/tds/assessment/Algorithm.java

```diff
@@ -0,0 +1,46 @@
+package tds.assessment;
+
+/**
+ * Enumeration representing assessment algorithm types
+ */
+public enum Algorithm {
+    ADAPTIVE_2("adaptive2"),
+    VIRTUAL("virtual"),
+    FIXED_FORM("fixedform");
+
+    private String algorithmName;
+
+    Algorithm(String algorithmName) {
+        this.algorithmName = algorithmName;
+    }
+
+    /**
+     * @return the name of the algorithm 
+     */
+    public String getAlgorithmName() {
+        return algorithmName;
+    }
+
+    @Override
+    public String toString() {
+        return this.getAlgorithmName();
+    }
+
+    /**
+     * Returns a {@link tds.assessment.Algorithm} enum for the algorithm name
+     *
+     * @param algorithmName
+     * @return
+     */
+    public static Algorithm fromAlgorithm(String algorithmName) {
```

Comment:
Suggest using `fromValue` or `fromAlgorithmName` depending on the result of my other comment.